src/commands/deploy/setPreDeployTaskForDotnet.ts

```diff
@@ -0,0 +1,112 @@
+        if (input === dontShowAgainButton) {
+            await updateWorkspaceSetting(configurePreDeployTasksSetting, false, context.workspace.uri.fsPath);
+        } else {
+            // resolves to "."if it is not a subfolder
+            const subfolder: string = path.dirname(path.relative(context.workspace.uri.fsPath, csprojFile));
+            const deploySubpath: string = path.posix.join(subfolder, 'bin', 'Release', 'publish');
+
+            await updateWorkspaceSetting(preDeployTaskSetting, 'publish', context.workspace.uri.fsPath);
+            await updateWorkspaceSetting(constants.configurationSettings.deploySubpath, deploySubpath, workspaceFspath);
+
+            // update the deployContext.deployFsPath with the .NET output path since getDeployFsPath is called prior to this
+            // purposely not using posix since this is happening at runtime
+            context.deployFsPath = path.join(context.workspace.uri.fsPath, deploySubpath);
+
+            const defaultTasks = getTasks(deploySubpath, subfolder);
+            tasks.updateTasks(context.workspace, defaultTasks);
+        }
+
+    }
+
+    async function tryGetCsprojFile(projectPath: string): Promise<string | undefined> {
+        let projectFiles: string[] = await checkFolderForCsproj(projectPath);
+        // it's a common pattern to have the .csproj file in a subfolder so check one level deeper
+        if (projectFiles.length === 0) {
+            for (const folder of fse.readdirSync(projectPath)) {
+                const filePath: string = path.join(projectPath, folder);
+                if (fse.existsSync(filePath) && (await fse.stat(filePath)).isDirectory()) {
+                    projectFiles = projectFiles.concat(await checkFolderForCsproj(filePath));
+                }
+            }
+        }
+
+        context.telemetry.properties.numOfCsprojFiles = projectFiles.length.toString();
+
+        // if multiple csprojs were found, ignore them
+        return projectFiles.length === 1 ? projectFiles[0] : undefined;
+
+        async function checkFolderForCsproj(filePath: string): Promise<string[]> {
+            const files: string[] = fse.readdirSync(filePath);
+            const filePaths: string[] = files.map((f: string) => {
+                return path.join(filePath, f);
+            });
+
+            return filePaths.filter((f: string) => /\.csproj$/i.test(f));
+        }
+    }
+}
+
+function getTasks(deploySubpath: string, subfolder: string): TaskDefinition[] {
+    // tslint:disable-next-line: no-unsafe-any no-invalid-template-strings
+    const cwd: string = path.posix.join('${workspaceFolder}', subfolder);
+    return [
+        {
+            label: 'clean',
+            command: 'dotnet clean',
+            type: 'shell',
+            problemMatcher: '$msCompile',
+            options: {
+                cwd
+            }
+        },
+        {
+            label: 'publish',
```

Comment:
I recommend you follow these steps:
1. Create a new folder
1. Run `dotnet new webapp`
1. Open that folder in VS Code
1. Wait a bit for this notification
![Screen Shot 2019-08-29 at 2 05 09 PM](https://user-images.githubusercontent.com/11282622/63976486-4e430200-ca66-11e9-8988-8ad3457bbb87.png)
1. Click yes

You'll notice that it adds a `publish` task already. Shouldn't we be using what they created? Seems odd to have two different publish tasks

Also their dialog isn't perfect, but I like it better than ours. It feels more like plain english instead of technical terms/settings.

Reply:
ping @fiveisprime can you also go through this experience and help Nathan decide how his work should integrate with the basic C# experience

Reply:
What about in the case of a csproj in a subfolder?

The publish task that they create seems to have some extra args that make the logs less verbose?  It also doesn't dependOn the clean task.

I think I would rather write over it.

Reply:
Lol I trust the C# extension for this kind of thing more than I trust us. We can also file issues on them if we truly think something is better from ours

For example - is the clean task actually necessary? Maybe publish cleans by default

Reply:
We should rely on the C# extension as much as possible. 

1. They know best how to manage the tasks and what's necessary
1. I believe we're starting to creep out of scope of the app service extension

It seems safe to assume that C# customers will have the extension installed so I'm not so worried about that. I'll schedule a meeting with the C# extension owners next week to work through what we should do and what they might be able to add to help us.

Reply:
Based on the meeting we had with Ravi, I would say our takeaway is that we shouldn't rely on the C# extension for the publish task.  Apparently, they haven't changed their task generator in a while and have no plans to change it.  But I will change our tasks to be formatted in the same way as theirs.

An issue that came up from the meeting is if we don't replace the user's publish task, we can't guarantee that we know where the output bits will be.  We might be able to read `dotnet publish`'s output and find the folder that way?

Reply:
> Based on the meeting we had with Ravi, I would say our takeaway is that we shouldn't rely on the C# extension for the publish task. 

I feel like we're still relying on the C# extension and it's publish task, but perhaps we have different definitions of "rely" so I just wanted to clarify what I expect:
1. If the publish task exists (which it will in most cases IMO) we should use that
1. If the publish task doesn't exist, we would create a task for them, but it will basically be a duplicate of what the C# extension would've created

> An issue that came up from the meeting is if we don't replace the user's publish task, we can't guarantee that we know where the output bits will be.

I feel like this feature is more about the 90% case. Users can always change these settings themselves, so I'm not too worried about guaranteeing the default settings will work 100% of the time.